### ISSUE TITLE
[java] false-positive in DD-part of DataflowAnomalyAnalysis

### ISSUE THREAD

**Issue by Vampire** [pmd/pmd#1675]
**Affects PMD Version:** *6.11.0*
**Rule:** [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis)
**Description:**
`formatter` is reported in the snippet below.
But without the explicit initializer, the first line in the try-catch could throw and the variable would be undefined which is also why the compiler wouldn't allow that, so this rule shouldn't report it either.

**Code Sample demonstrating the issue:**

``` java
public class Test {
    public void test() {
        Formatter formatter = null;
        if (formatterClassName != null) {
            try {
                Class<? extends Formatter> formatterClass = findClass(formatterClassName);
                formatter = formatterClass.getDeclaredConstructor().newInstance();
            } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InstantiationException | InvocationTargetException ignored) {
            }
        }
        setFormatter((formatter == null) ? new SimpleFormatter() : formatter);
    }
}
```

Result:
`n/a:3:	Found 'DD'-anomaly for variable 'formatter' (lines '3'-'7').`

**Running PMD through:** *Gradle*

**Comment by adangel** [comment]
The rule [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis) has been deprecated in favour of [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment), which is available with PMD 6.26.0.

Therefore this issue wont be fixed.

See also #2647.
